Assets/Scripts/Bomb.cs

```diff
@@ -0,0 +1,39 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Bomb : Enemy {
+    // These are only initial values for behaviour construction
+    private const float _MAX_SPEED = 4;
+    private const float _ACCELERATION_RATE = 1;
+
+    private Rigidbody2D _player;
+
+    // Bomb does nothing while idle, approaches player otherwise
+    protected override void Start() {
+        this._myRigidbody = this.GetComponent<Rigidbody2D>();
+        this._player = GameObject.FindGameObjectWithTag("Player").GetComponent<Rigidbody2D>();
+
+        // Uncomment this to check InvalidEnemyStateException checking
+        //this._idleBehaviour = new ApproachTarget(this, _player, Bomb._MAX_SPEED, Bomb._ACCELERATION_RATE, ENEMY_STATE.idle);
+
+        this._idleBehaviour = new DoNothing(this, ENEMY_STATE.idle);
+        this._awareBehaviour = new ApproachTarget(this, this._player, Bomb._MAX_SPEED, Bomb._ACCELERATION_RATE, ENEMY_STATE.aware);
+        this._engagedBehaviour = new ApproachTarget(this, this._player, Bomb._MAX_SPEED, Bomb._ACCELERATION_RATE, ENEMY_STATE.engaged);
+    }
+
+    // Bomb aware of player within 100 units, engaged within 50.
+    protected override void ChangeState() {
```

Comment:
I think we should instead name this something like `GetState` or `UpdateState` so as not imply that the outgoing state is guaranteed to be different from the original state